### PR TITLE
cephadm: respect --skip-firewalld flag

### PR DESCRIPTION
**Pull request by adk3798**
Fixes: https://tracker.ceph.com/issues/54137

Signed-off-by: Adam King <adking@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by mgfritch — APPROVED** [review]
couple of nits, otherwise lgtm!

**Comment by adk3798** [comment]
http://pulpito.front.sepia.ceph.com/adking-2022-03-21_05:58:33-orch:cephadm-wip-adk-testing-2022-03-20-2156-distro-basic-smithi/

1 Dead job due to failing to power on a machine
2 Failed jobs due to unknown flag: --sort caused by another PR in the run
1 Failed job due to Error EINVAL: Invalid service 'ingress.nfs.test'. Use 'ceph orch ls' to list available services caused by another PR in the run

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins test api

File: src/cephadm/tests/test_cephadm.py [hunk 1]
```diff
@@ -214,10 +215,42 @@ def wrap_test(address, expected):
         for address, expected in tests:
             wrap_test(address, expected)
 
+    @mock.patch('cephadm.Firewalld', mock_bad_firewalld)
+    @mock.patch('cephadm.logger')
+    @mock.patch('os.fchown')
+    def test_skip_firewalld(self, fchown, logger):
```
[the remark this answers is not in the document]
Comment (adk3798) [reply src/cephadm/tests/test_cephadm.py]: done

File: src/cephadm/tests/test_cephadm.py [hunk 2]
```diff
@@ -214,10 +215,42 @@ def wrap_test(address, expected):
...
+        # test --skip-firewalld actually skips changing firewall
```
[the remark this answers is not in the document]
Comment (adk3798) [reply src/cephadm/tests/test_cephadm.py]: is what's there now what you meant?

**Comment by adk3798** [comment]
http://pulpito.front.sepia.ceph.com/adking-2022-03-23_02:54:35-orch:cephadm-wip-adk-testing-2022-03-22-2000-distro-basic-smithi/

2 Failures caused by wrong error code from host add command due to another PR included in the run

**Pull request merged**